src/storage/trie/merkle_trie.rs

```diff
@@ -128,7 +129,16 @@ impl MerkleTrie {
 
     pub fn commit(&self) -> Result<(), HubError> {
         if let Some(root) = self.root.write().unwrap().as_mut() {
-            self.unload_from_memory(root, true)
+            self.unload_from_memory(root)
```

Comment:
The DB commit should happen outside the trie, and maybe the trie should just always unload from memory?

This is going to cause performance issues, we'll likely need some kind of LRU cache for frequently accessed fids. But we can worry about that later.